R/derive_vars_extreme_date.R

```diff
@@ -0,0 +1,154 @@
+#'                           time_imputation = "first",
+#'                           flag_imputation = "AUTO",
+#'                           min_dates = NULL,
+#'                           max_dates = NULL,
+#'                           mode="LAST") %>%
+#'   select(USUBJID, TRTEDTM)
+derive_vars_extreme_date <- function(dataset,
+                                     subject_keys =vars(STUDYID,USUBJID),
+                                     dataset_date,
+                                     filter_dataset_date = NULL,
+                                     dtc,
+                                     order,
+                                     new_var,
+                                     date_imputation,
+                                     time_imputation,
+                                     flag_imputation = "AUTO",
+                                     min_dates = NULL,
+                                     max_dates = NULL,
+                                     mode) {
+
+  new_var <- assert_symbol(enquo(new_var))
+  dtc <- assert_symbol(enquo(dtc))
+  assert_data_frame(dataset, subject_keys)
+  assert_data_frame(dataset_date, required_vars = quo_c(subject_keys, order))
+  assert_character_scalar(mode, values = c("first", "last"), case_sensitive = F)
+
+  filter_dataset_date <- assert_filter_cond(enquo(filter_dataset_date), optional = TRUE)
+  filter_dataset_date <- enquo(filter_dataset_date)
+
+  assert_character_scalar(flag_imputation, values = c("auto", "both", "date", "time", "none"),
+                          case_sensitive = FALSE)
+
+  #Issue warning that new_var should be DTM
+  if (str_detect(rlang::quo_name(new_var), "DTM", negate = T)) {
+    msg <- sprintf(
+      "This function derives the treatment datetime, new_var must end in DTM.")
+    warn(msg)
+  }
+
+  add <- dataset_date %>%
+    mutate(imputed_dtc = convert_dtc_to_dtm(!!dtc,
+                                            date_imputation = date_imputation,
+                                            time_imputation = time_imputation,
+                                            min_dates = min_dates,
+                                            max_dates = max_dates)
+    )
+
+  #filter EX data based on study requirements
+  if (quo_is_null(filter_dataset_date)) {
+    add
+    msg <- sprintf(
+      "Input EX dataset has no been filtered. Check study requirement if ex needs to be filtered.")  # nolint
+    inform(msg)
+  } else {
+    add <- add %>% filter(!!filter_dataset_date)
+  }
+
+  add <- add %>%
+    filter_extreme(order = order,
+                   by_vars = subject_keys,
+                   mode = mode) %>%
+    mutate(!!new_var := imputed_dtc) %>%
+    select(!!!subject_keys, !!new_var, !!dtc, imputed_dtc)
+
+  #add imputation flags
+  if (flag_imputation %in% c("both", "date") ||
+      flag_imputation == "auto" && !is.null(date_imputation)) {
+    # add --DTF if not there already
+    dtf <- paste0("TRT", "DTF")
```

Comment:
The name of the date imputation flag should not be hard-coded.

Reply:
good catch